internal/pkg/skuba/replica/replica.go

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package replica
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+)
+
+const (
+	highAvailabilitylabel = "caasp.suse.com/skuba-replica-ha"
+	maxSize               = 3
+	minSize               = 2
+	patchAffinityRemove   = `[
+		{
+			"op":"remove",
+			"path":"/spec/template/spec/affinity"
+		}
+	]`
+	patchAffinityRequired = `{
+		"spec": {
+			"template": {
+				"spec": {
+					"affinity": {
+						"podAntiAffinity": {
+							"requiredDuringSchedulingIgnoredDuringExecution": [
+								{
+									"labelSelector": {
+										"matchExpressions": [
+											{
+												"key": "app",
+												"operator": "In",
+												"values": [
+													"%s"
+												]
+											}
+										]
+									},
+									"topologyKey": "kubernetes.io/hostname"
+								}
+							]
+						}
+					}
+				}
+			}
+		}
+	}`
+	patchReplicas = `{
+		"spec": {
+			"replicas": %v
+		}
+	}`
+)
+
+// Helper provide methods for replica update of deployment
+// resource that has highAvailabilitylabel.
+type Helper struct {
+	Client            clientset.Interface
+	SelectDeployments appsv1.DeploymentList
+	MaxSize           int
+	MinSize           int
+	ClusterRoleSize   int
+}
+
+// NewHelper creates a helper to update deployment replicas.
+func NewHelper(client clientset.Interface) (*Helper, error) {
+	fromSelectors, err := client.AppsV1().Deployments(metav1.NamespaceSystem).List(
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=true", highAvailabilitylabel),
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	cp, err := kubernetes.GetControlPlaneNodes(client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Helper{
+		Client:            client,
+		ClusterRoleSize:   len(cp.Items),
+		MaxSize:           maxSize,
+		MinSize:           minSize,
+		SelectDeployments: *fromSelectors,
```

Comment:
suggest `SelectDeployments` as a function to call at runtime.

Reply:
After discussing with Chinya, this is optional for this case.

Reply:
No change required after discussion.